R/error-checks.R

```diff
@@ -42,8 +42,27 @@ isOfType <- function(object, type, nullAllowed = FALSE) {
 
 #' Check if a vector of values is included in another vector of values
 #'
-#' @param values A single value or a vector of values.
-#' @param parentValues A single value or a vector of values.
+#' @param values A vector of values.
+#' @param parentValues A vector of values where `values` are checked for
+#'   inclusion.
+#'
+#' @description
+#'
+#' A helper function to check if a vector of values is included in another
+#' vector of values.
+#'
+#' Note that arguments can also be of length `1` (e.g. `values = "x"`). But
+#' whether this constitutes a vector, and thus a valid function argument,
+#' depends on the **class** of the object. For all built-in classes in R
+#' (numeric, character, logical, etc.), a single value is a valid vector (e.g.
+#' `"x"` is a vector of character type, as shown by `is.vector("x")`).
+#'
+#' But, for other classes (like `R6`), you will first need to convert a single
+#' value to a vector. For example, if `myClass` object is an instance of `R6`
+#' class, you can't enter `values = myClass`, since a single `R6` object is not
+#' a vector. You will instead need to use `values = c(myClass)` or `values =
```

Comment:
@IndrajeetPatil Can I ask why we are not doing sthg like values <-c (values) if is.vector(values) is false. That way we always have a vector and this would work even with a myClass as single argument

Reply:
We can definitely do that!

I just wasn't sure if that would cause any adverse effects in places where this is used. The original issue was more about documentation, and so I refrained from making any change that would modify function behaviour.

Reply:
I see. It just feels that we are documenting a behavior that is unexpected to me instead of having a consistent behavior. I am not sure where this is being used or where is this method coming from.
@pchelle maybe has some insights?

Reply:
https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/search?q=isIncluded

seems to only be with strings or primitive types

Reply:
Done! Now the behavior is consistent :)